weblate/trans/models/component.py

```diff
@@ -770,6 +783,7 @@ class Component(models.Model, PathMixin, CacheKeyMixin, ComponentCategoryMixin):
     remote_revision = models.CharField(max_length=200, default="", blank=True)
     local_revision = models.CharField(max_length=200, default="", blank=True)
     processed_revision = models.CharField(max_length=200, default="", blank=True)
+    to_delete = models.BooleanField(default=False)
```

Comment:
This would need a database index. Otherwise it will make the things unnecessarily slow.